test/load_test.c

```diff
@@ -0,0 +1,294 @@
+#include "../src/def/game_defs.h"
+#include "../src/def/game_history_defs.h"
+#include "../src/ent/game_history.h"
+#include "../src/impl/config.h"
+#include "../src/impl/load.h"
+#include "../src/util/io_util.h"
+#include "../src/util/string_util.h"
+#include "test_util.h"
+#include <assert.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+void test_cross_tables_integration(void) {
+  // Only run if MAGPIE_INTEGRATION_TESTS environment variable is set
+  if (!getenv("MAGPIE_INTEGRATION_TESTS")) {
+    printf("Skipping cross-tables integration test (set MAGPIE_INTEGRATION_TESTS=1 to enable)\n");
+    return;
+  }
+
+  GameHistory *game_history = game_history_create();
+  ErrorStack *error_stack = error_stack_create();
+
+  // Test with a known working cross-tables game URL
+  // Using a classic game that should be stable on cross-tables
+  Config *config = config_create_default_with_data_paths(error_stack, "data");
+  
+  if (!error_stack_is_empty(error_stack)) {
+    printf("Skipping cross-tables integration test - data files not available\n");
+    error_stack_destroy(error_stack);
+    game_history_destroy(game_history);
+    return;
+  }
+  
+  DownloadGCGOptions options = {
+    .source_identifier = "https://www.cross-tables.com/annotated.php?u=54938#0%23",
+    .lexicon = NULL,
+    .config = config
+  };
+
+  printf("Testing cross-tables download with URL: %s\n", options.source_identifier);
+  
+  download_gcg(&options, game_history, error_stack);
+
+  if (!error_stack_is_empty(error_stack)) {
+    printf("Cross-tables integration test failed: ");
+    error_stack_print_and_reset(error_stack);
+    // Don't assert fail here since network issues are common
+    printf("(This may be due to network connectivity or changes to cross-tables.com)\n");
+  } else {
+    // Basic validation that we got a valid game
+    assert(game_history_get_number_of_events(game_history) > 0);
+    
+    // Check that we have players
+    assert(game_history_both_players_are_set(game_history));
+    
+    printf("Cross-tables integration test passed: loaded %d game events\n",
+           game_history_get_number_of_events(game_history));
+  }
```

Comment:
This section (and similar sections below) is a good candidates for refactoring out into a new test helper function, something like "validate_download_gcg" which takes just the options as an argument and asserts that:

 - error stack is empty
 - game history has some events
 - both player names are set

You can create and destroy the game history and error stack in this new function so you don't have to do it every time for each test.

Reply:
You could also make this function handle the error cases as well by passing in the expected error stack error enum and asserting that as well.